browser/src/layer/tile/CanvasTileLayer.js

```diff
@@ -5297,7 +5325,64 @@ L.CanvasTileLayer = L.Layer.extend({
 		return ctx;
 	},
 
-	_applyDelta: function(tile, rawDelta, isKeyframe, wireMessage) {
+	_applyCompressedDelta: function(tile, rawDelta, isKeyframe, wireMessage) {
+		if (window.Worker) {
+			var e = { tile: tile, rawDelta: rawDelta, isKeyframe: isKeyframe, wireMessage: wireMessage };
+			if (tile.hasPendingUpdate() || !this._workerBuffers.length) {
+				this._workerPendingEvents.push(e);
+				return;
+			}
+
+			var buffer = this._workerBuffers.shift();
+			var id = this._workerId++;
+
+			try {
+				this._worker.postMessage(
+					{ 'message': 'decompress',
+					  'tile': { id: id,
+								rawDelta: rawDelta, // Note: This is copied intentionally
```

Comment:
It's a shame that we have to copy the array here, but rawDelta is often a subarray and not doing so leads to bad things happening sometimes :) I wonder if we could re-jig things so that we could avoid this copy?